backbone.js

```diff
@@ -746,6 +746,15 @@
   var setOptions = {add: true, remove: true, merge: true};
   var addOptions = {add: true, remove: false};
 
+  // Splices `insert` into `array` at index `at`.
+  var splice = function(array, insert, at) {
+    var tail = Array(array.length - at);
+    var length = insert.length;
+    for (var i = 0; i < tail.length; i++) tail[i] = array[i + at];
+    for (i = 0; i < length; i++) array[i + at] = insert[i];
+    for (i = 0; i < tail.length; i++) array[i + length + at] = tail[i];
+  };
```

Comment:
Proof of speed?

On Mon, Aug 3, 2015 at 6:45 PM, Justin Ridgewell <notifications@github.com>
wrote:

> In backbone.js
> <https://github.com/jashkenas/backbone/pull/3663#discussion_r36140323>:
>
> > @@ -746,6 +746,15 @@
> >    var setOptions = {add: true, remove: true, merge: true};
> >    var addOptions = {add: true, remove: false};
> >
> > +  // Splices `insert` into `array` at index `at`.
> > +  var splice = function(array, insert, at) {
> > +    var tail = Array(array.length - at);
> > +    var length = insert.length;
> > +    for (var i = 0; i < tail.length; i++) tail[i] = array[i + at];
> > +    for (i = 0; i < length; i++) array[i + at] = insert[i];
> > +    for (i = 0; i < tail.length; i++) array[i + length + at] = tail[i];
> > +  };
>
> Speed. [image: :racehorse:] (How is there not a racecar emoji?) See the
> OP.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jashkenas/backbone/pull/3663/files#r36140323>.
>